VirtoCommerce.CatalogModule.Web/Scripts/blades/bulk/action-list.js

```diff
@@ -16,22 +16,29 @@ angular.module('virtoCommerce.catalogModule')
         $scope.openBlade = function(data) {
             var newBlade = {};
             angular.copy(data, newBlade);
-            newBlade.selectedCategories = blade.selectedCategories;
-            newBlade.selectedProducts = blade.selectedProducts;
-            newBlade.catalog = blade.catalog;
 
-            if (angular.isFunction(data.onInitialize)) {
-                data.onInitialize(newBlade);
-            }
+            var registrationInfo = bulkActionServiceRegistrar.getByName(data.name);
+            if (!registrationInfo) {
+                bladeNavigationService.setError(`Can't find controller for action ${data.name}`, blade);
+            } else {
+                newBlade.controller = registrationInfo.controller;
+                newBlade.template = registrationInfo.template;
+                newBlade.actionDataContext.dataQuery = angular.extend({
```

Comment:
I think it could be replaced with:
```
               angular.extend(newBlade.actionDataContext.dataQuery, {
                    categoryIds: _.pluck(blade.selectedCategories, 'id'),
                    objectIds: _.pluck(blade.selectedProducts, 'id'),
                    catalogIds: [blade.catalog.id]
                });
```